Reviewers API responds with a 500 error when trying to view a file with blank spaces in its name

### Describe the problem and steps to reproduce it:
1. Send the following request to the reviewers/browse API:
```
https://addons-dev.allizom.org/api/v5/reviewers/addon/618629/versions/1690063/?file=content script.js
```

### What happened?
The API responds with a 500 server error because the file name contains a blank space

### What did you expect to happen?
No server error

### Anything else we should know?
- the issue is also reproducible for the compare API and will completely prevent the compare page from loading in code-manager:

https://code.addons-dev.allizom.org/en-US/compare/618629/versions/1690062...1690063/

Update: - the issue can also happen if a folder name contains spaces - see [example](https://code.addons-dev.allizom.org/en-US/browse/618821/versions/1690254/?path=more%20rules%2Fmore%20more%20rules%2Fmore%283%29%20rules%2Fcoinhive.js)

The issue here is with the cache-key generation:

```python
E   memcache.Client.MemcachedKeyCharacterError: Control/space characters not allowed (key=b'amo:dfde4542a94b489990dc5ebb143711fd::1:reviewers:fileentriesserializer:hashes:f7f378aa9a95090bf858e32dfc8136d9c2bd7c36:content script.js')
```

--- a/src/olympia/reviewers/serializers.py
+++ b/src/olympia/reviewers/serializers.py
@@ -96,11 +96,13 @@ class FileEntriesSerializer(FileSerializer):
         commit = self._get_commit(obj)
         tree = self.repo.get_root_tree(commit)
 
-        selected_file_key = make_key(
-            selected_file, with_locale=False, normalize=True)
-        cache_key = (
+        # Normalize the key as we want to avoid that we exceed max
+        # key lengh because of selected_file.
+        cache_key = make_key(
             f'reviewers:fileentriesserializer:hashes'
-            f':{commit.hex}:{selected_file_key}')
+            f':{commit.hex}:{selected_file}',
+            with_locale=False,
+            normalize=True)
 
         def _calculate_hash():
             try:

--- a/src/olympia/reviewers/serializers.py
+++ b/src/olympia/reviewers/serializers.py
@@ -96,11 +96,11 @@ class FileEntriesSerializer(FileSerializer):
         commit = self._get_commit(obj)
         tree = self.repo.get_root_tree(commit)
 
-        cache_key = make_key(
+        selected_file_key = make_key(
+            selected_file, with_locale=False, normalize=True)
+        cache_key = (
             f'reviewers:fileentriesserializer:hashes'
-            f':{commit.hex}:{selected_file}',
-            with_locale=False,
-            normalize=True)
+            f':{commit.hex}:{selected_file_key}')
 
         def _calculate_hash():
             try:

--- a/src/olympia/reviewers/serializers.py
+++ b/src/olympia/reviewers/serializers.py
@@ -31,6 +31,7 @@ from olympia.files.models import File
 from olympia.reviewers.models import CannedResponse
 from olympia.versions.models import Version
 from olympia.lib.git import AddonGitRepository, get_mime_type_for_blob
+from olympia.lib.cache import make_key
 from olympia.lib import unicodehelper
 
 
@@ -95,9 +96,11 @@ class FileEntriesSerializer(FileSerializer):
         commit = self._get_commit(obj)
         tree = self.repo.get_root_tree(commit)
 
-        cache_key = (
+        cache_key = make_key(
             f'reviewers:fileentriesserializer:hashes'
-            f':{commit.hex}:{selected_file}')
+            f':{commit.hex}:{selected_file}',
+            with_locale=False,
+            normalize=True)
 
         def _calculate_hash():
             try:

--- a/src/olympia/reviewers/tests/test_views.py
+++ b/src/olympia/reviewers/tests/test_views.py
@@ -6086,6 +6086,38 @@ class TestReviewAddonVersionViewSetDetail(
         response = self.client.get(self.url + '?file=UNKNOWN_FILE')
         assert response.status_code == 404
 
+    def test_requested_file_contains_whitespace(self):
+        new_version = version_factory(
+            addon=self.addon, file_kw={'filename': 'webextension_no_id.xpi'})
+
+        repo = AddonGitRepository.extract_and_commit_from_version(new_version)
+
+        apply_changes(
+            repo, new_version, u'(function() {})\n', u'content script.js')
+
+        user = UserProfile.objects.create(username='reviewer')
+        self.grant_permission(user, 'Addons:Review')
+        self.client.login_api(user)
+
+        url = reverse_ns('reviewers-versions-detail', kwargs={
+            'addon_pk': self.addon.pk,
+            'pk': new_version.pk})
+
+        response = self.client.get(url + '?file=content script.js')
+        assert response.status_code == 200
+        result = json.loads(response.content)
+
+        assert result['file']['content'] == '(function() {})\n'
+
+        # make sure the correct download url is correctly generated
+        assert result['file']['download_url'] == absolutify(reverse(
+            'reviewers.download_git_file',
+            kwargs={
+                'version_id': new_version.pk,
+                'filename': 'content script.js'
+            }
+        ))
+
     def test_supports_search_plugins(self):
         self.addon = addon_factory(
             name=u'My Addôn', slug='my-addon',
@@ -6834,6 +6866,33 @@ class TestReviewAddonVersionCompareViewSet(
         change = result['file']['diff']['hunks'][0]['changes'][0]
 
         assert change['content'] == '# beastify'
+        assert change['type'] == 'insert'
+
+    def test_requested_file_contains_whitespace(self):
+        new_version = version_factory(
+            addon=self.addon, file_kw={'filename': 'webextension_no_id.xpi'})
+
+        repo = AddonGitRepository.extract_and_commit_from_version(new_version)
+
+        apply_changes(
+            repo, new_version, u'(function() {})\n', u'content script.js')
+
+        user = UserProfile.objects.create(username='reviewer')
+        self.grant_permission(user, 'Addons:Review')
+        self.client.login_api(user)
+
+        url = reverse_ns('reviewers-versions-compare-detail', kwargs={
+            'addon_pk': self.addon.pk,
+            'version_pk': self.version.pk,
+            'pk': new_version.pk})
+
+        response = self.client.get(url + '?file=content script.js')
+        assert response.status_code == 200
+        result = json.loads(response.content)
+        change = result['file']['diff']['hunks'][0]['changes'][0]
+
+        assert result['file']['diff']['path'] == 'content script.js'
+        assert change['content'] == '(function() {})'
         assert change['type'] == 'insert'
 
     def test_supports_search_plugins(self):

--- a/src/olympia/reviewers/tests/test_views.py
+++ b/src/olympia/reviewers/tests/test_views.py
@@ -6093,7 +6093,7 @@ class TestReviewAddonVersionViewSetDetail(
         repo = AddonGitRepository.extract_and_commit_from_version(new_version)
 
         apply_changes(
-            repo, new_version, u'(function() {})\n', u'content script.js')
+            repo, new_version, '(function() {})\n', 'content script.js')
 
         user = UserProfile.objects.create(username='reviewer')
         self.grant_permission(user, 'Addons:Review')
@@ -6875,7 +6875,7 @@ class TestReviewAddonVersionCompareViewSet(
         repo = AddonGitRepository.extract_and_commit_from_version(new_version)
 
         apply_changes(
-            repo, new_version, u'(function() {})\n', u'content script.js')
+            repo, new_version, '(function() {})\n', 'content script.js')
 
         user = UserProfile.objects.create(username='reviewer')
         self.grant_permission(user, 'Addons:Review')

